src/VisualStudio/Core/Def/Progression/GraphProvider.cs

```diff
@@ -176,10 +176,12 @@ public IEnumerable<GraphCommand> GetCommands(IEnumerable<GraphNode> nodes)
     {
```

Comment:
really dumb question.  is the IENumerable passed in here lazy?  If so, can we def make it *non* lazy (an immutable-array would be good).  I would hate if each .Any was lazily evaluating that underlying list again and again.

Reply:
oh.  we don't even own this list.  

*consider* acutally making a single loop that assigns into locals (or a helper function for this).  having one iteration would make me feel a lot more confident about this.

Reply:
OK, I modified the code to do a single loop for most of the data needed. I didn't have that loop calculate all the data as some of it is only needed for certain symbolkind/typekind combinations. The new (artificial) numbers still look good:

```
s_swOld.ElapsedMilliseconds	1939	long
s_swNew.ElapsedMilliseconds	426	long
```